File: Selenium/Project/CRM_Activity7.py
# Reading additional information
# Goal: Reading a popup that contains additional information about the account/lead.
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

# Start the Driver
with webdriver.Chrome() as driver:

   driver.get("https://alchemy.hguy.co/crm/")
 
# Find the username field
   username = driver.find_element(By.ID, "user_name")
# Find the password field
   password = driver.find_element(By.ID, "username_password")

    # Enter the given credentials
    # Enter username
   username.send_keys("admin")
    # Enter password
   password.send_keys("pa$$w0rd")

    # Find the login button
   login = driver.find_element(By.XPATH, "//input[@type='submit']")
   login.click()

    #Function
   driver.maximize_window()
   time.sleep(2)

   salesMenu = driver.find_element(By.ID, "grouptab_0")
   salesMenu.click()
    
   leadsMenu = driver.find_element(By.ID, "moduleTab_9_Leads")
   leadsMenu.click()
   time.sleep(2)
   addDetIcon = driver.find_element(By.XPATH, "//span[@title='Additional Details']").click()
   time.sleep(2)
   mobileNumb =driver.find_element(By.XPATH, "//span[@class='phone']").text
   print("The mobile number is :" + mobileNumb)

File: Selenium/Project/CRM_Activity7.java
package Project;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class CRM_Activity7 {
	WebDriver driver;
	WebDriverWait wait;
    
    @BeforeClass
    public void setUp() {
        //Open a browser
        driver = new ChromeDriver();
        driver.get("http://alchemy.hguy.co/crm");
    }
    
    @Test(priority = 1)
	  public void loginFormTest() {
		  //Provide username and password
		  driver.findElement(By.id("user_name")).sendKeys("admin");
		  driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
		  //Find the submit button and click it
		  driver.findElement(By.xpath("//input[@type='submit']")).click();
		  
		  
    }
    
    @Test(priority = 2)
	public void additionalInformation() throws InterruptedException {
		driver.manage().window().maximize();
//		Navigate to Sales -> Leads
		driver.findElement(By.id("grouptab_0")).click();
		driver.findElement(By.id("moduleTab_9_Leads")).click();
		Thread.sleep(2000);
//		In the table, find the Additional information icon at the end of the row of the lead information. Click it.
		driver.findElement(By.xpath("//span[@title='Additional Details']")).click();
		Thread.sleep(2000);
//		Read the popup and print the phone number displayed in it.
		WebElement additionalDetails = driver.findElement(By.xpath("//span[@class='phone']"));
		String mobileNumber = additionalDetails.getText();
		System.out.println("The mobile number is :" + mobileNumber);
		Assert.assertFalse(mobileNumber.isEmpty(),"Mobile Number not available");

	}

    @AfterClass
    public void tearDown() {
        driver.quit();
    }
    
}
